tools/bundle_macos_universal_wheels.py

```diff
@@ -0,0 +1,362 @@
+    filename_parts = first_wheel_info["filename"].split("-")
+
+    # Find the oldest macOS version among the wheels for the universal2 wheel
+    min_version = None
+    for wheel_data in matching_wheels:
+        wheel_info = wheel_data["wheel_info"]
+        arch, version = parse_macos_platform(wheel_info["platform_tag"])
+        if min_version is None or version < min_version:
+            min_version = version
+
+    # Replace the platform tag with universal2 using the oldest version
+    major, minor = min_version
+    filename_parts[-1] = f"macosx_{major}_{minor}_universal2.whl"
+
+    return "-".join(filename_parts)
+
+
+async def merge_wheels(
+    session: aiohttp.ClientSession,
+    package_name: str,
+    matching_wheels: list[dict[str, Any]],
+    universal_wheels_dir: Path,
+) -> bool:
+    """Download wheels and merge them using delocate-merge.
+
+    Returns True if successful (including if already exists), False if failed.
+    """
+
+    # Calculate expected merged wheel filename
+    expected_filename = calculate_merged_wheel_name(matching_wheels)
+    expected_path = universal_wheels_dir / expected_filename
+
+    # Skip if merged wheel already exists
+    if expected_path.exists():
+        print(f"  ✓ Already exists: {expected_filename}")
+        return True
+
+    # Create temporary directory for downloads
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_path = Path(temp_dir)
+
+        try:
+            # Download both wheels concurrently
+            download_tasks = [download_wheel(session, wheel_data["url"], temp_path) for wheel_data in matching_wheels]
+            wheel_paths = await asyncio.gather(*download_tasks)
+        except Exception as e:
+            print(f"  ✗ Failed to download wheels for merging: {e}")
+            return False
+
+        # Execute delocate-merge to output directory
+        try:
+            subprocess.run(
+                ["delocate-merge", "-w", str(universal_wheels_dir), str(wheel_paths[0]), str(wheel_paths[1])],
+                check=True,
+                capture_output=True,
+                text=True,
+            )
+
+            # Check if the expected file was created
+            if expected_path.exists():
+                print(f"  ✓ Created: {expected_filename}")
+                return True
+            else:
+                print(f"  ✗ Expected file not created: {expected_filename}")
+                return False
+
+        except subprocess.CalledProcessError as e:
+            print(f"  ✗ delocate-merge failed: {e}")
+            if e.stdout:
+                print(f"    stdout: {e.stdout}")
+            if e.stderr:
+                print(f"    stderr: {e.stderr}")
+            return False
+
+
+def is_pure_python_wheel(wheel_info: dict) -> bool:
+    return wheel_info["abi_tag"] == "none" and wheel_info["platform_tag"] == "any" and wheel_info["python_tag"] == "py3"
+
+
+async def main() -> int:
```

Comment:
is the tldr of this file
- go through all packages
- search on pypi whether universal wheels exist
  - if yes install those
- if no, try if we can build a universal wheel / fetch intel and arm wheels
- put all the new wheels into specific folder

?

Reply:
yeah. 

For building an universal executable we need:
- An universal python executable, we have that
- Every installed binary wheel needs to be universal2